packages/common/src/pipes/keyvalue_pipe.ts

```diff
@@ -50,31 +50,30 @@ export class KeyValuePipe implements PipeTransform {
   private differ!: KeyValueDiffer<any, any>;
   private keyValues: Array<KeyValue<any, any>> = [];
 
-  transform<K, V>(input: null, compareFn?: (a: KeyValue<K, V>, b: KeyValue<K, V>) => number): null;
-  transform<V>(
-      input: {[key: string]: V}|ReadonlyMap<string, V>,
-      compareFn?: (a: KeyValue<string, V>, b: KeyValue<string, V>) => number):
-      Array<KeyValue<string, V>>;
-  transform<V>(
-      input: {[key: string]: V}|ReadonlyMap<string, V>|null,
-      compareFn?: (a: KeyValue<string, V>, b: KeyValue<string, V>) => number):
-      Array<KeyValue<string, V>>|null;
-  transform<V>(
-      input: {[key: number]: V}|ReadonlyMap<number, V>,
-      compareFn?: (a: KeyValue<number, V>, b: KeyValue<number, V>) => number):
-      Array<KeyValue<number, V>>;
-  transform<V>(
-      input: {[key: number]: V}|ReadonlyMap<number, V>|null,
-      compareFn?: (a: KeyValue<number, V>, b: KeyValue<number, V>) => number):
-      Array<KeyValue<number, V>>|null;
   transform<K, V>(
       input: ReadonlyMap<K, V>,
       compareFn?: (a: KeyValue<K, V>, b: KeyValue<K, V>) => number): Array<KeyValue<K, V>>;
+  transform<K extends number, V>(
+      input: Record<K, V>, compareFn?: (a: KeyValue<string, V>, b: KeyValue<string, V>) => number):
```

Comment:
Shouldn't this be
```suggestion
      input: Record<K, V>, compareFn?: (a: KeyValue<number, V>, b: KeyValue<number, V>) => number):
```

Which makes it equivalent to the next overload with `K extends string|number`.

Reply:
no, the `keyvalue` pipe passes the `number` keys as strings in the case of a `Record<number, V>` object :(

I tried to explain it in the commit message, but maybe it is not explicit enough (and maybe we should also add a comment in the code to explain it, so it does not get lost in the git history)